### PR TITLE
Document versioned endpoint behavior for PrivateLink setup

### PR DESCRIPTION
**Pull request by bobmshannon**
### What does this PR do?
Adds a note to clarify that additional configuration is not required for setting up PrivateLink with agents using versioned endpoints.

### Motivation

Make it easier to configure and setup PrivateLink.

### Preview

https://docs-staging.datadoghq.com/bobmshannon/privatelink_docs_enhancement/agent/guide/private-link

### Additional Notes

N/A

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/agent/guide/private-link.md [hunk 1]
```diff
@@ -173,6 +173,8 @@ This returns `metrics.agent.datadoghq.com`, the private hosted zone name that yo
 
     **Note**: To retrieve your DNS name, see the [View endpoint service private DNS name configuration documentation.][2]
 
+    **Note**: The agent sends telemetry to versioned endpoints e.g. `<version>-app.agent.datadoghq.com` which resolves to `metrics.agent.datadoghq.com` via a CNAME alias. Therefore, you only need to set up a private hosted zone for `metrics.agent.datadoghq.com`.
```
Comment (buraizu) [inline content/en/agent/guide/private-link.md:176]:
```suggestion
    **Note**: The Agent sends telemetry to versioned endpoints, for example, `<version>-app.agent.datadoghq.com` which resolves to `metrics.agent.datadoghq.com` through a CNAME alias. Therefore, you only need to set up a private hosted zone for `metrics.agent.datadoghq.com`.
```

**Review by buraizu — CHANGES_REQUESTED** [review]
Thanks for the PR, just left some minor suggestions per our [contributing guidelines](https://github.com/DataDog/documentation/blob/master/CONTRIBUTING.md).

I also noticed that the link in line 139 is not opening the expected page, but I opened a separate PR to address that:
https://github.com/DataDog/documentation/pull/16383

File: content/en/agent/guide/private-link.md [hunk 2]
```diff
@@ -173,6 +173,8 @@ This returns `metrics.agent.datadoghq.com`, the private hosted zone name that yo
 
     **Note**: To retrieve your DNS name, see the [View endpoint service private DNS name configuration documentation.][2]
```
Comment (buraizu) [inline content/en/agent/guide/private-link.md:175]:
```suggestion
    **Notes**: 
      - To retrieve your DNS name, see the [View endpoint service private DNS name configuration documentation.][2]

```

**Review by buraizu — CHANGES_REQUESTED** [review]
One additional edit to consolidate the notes into a single callout

File: content/en/agent/guide/private-link.md [hunk 3]
```diff
@@ -136,7 +136,7 @@ However, to route traffic to Datadog's PrivateLink offering in `us-east-1` from
 
 8. Click on the VPC endpoint ID to check its status.
 9. Wait for the status to move from _Pending_ to _Available_. This can take up to 10 minutes.
-10. Once this is created, route traffic from another region to send the data to Datadog using private link. For more information, see the [Work With VPC Peering connections][2] AWS documentation. 
+10. Once this is created, route traffic from another region to send the data to Datadog using private link. For more information, see the [Work With VPC Peering connections][2] AWS documentation.
```
[the remark this answers is not in the document]
Comment (bobmshannon) [reply content/en/agent/guide/private-link.md]: Updated the wording to match most of this suggestion. (PrivateLink is not used to route traffic from another region, but instead VPC peering.)

[the remark this answers is not in the document]
Comment (bobmshannon) [reply content/en/agent/guide/private-link.md]: How about this?
```
After creating the endpoint, use VPC peering to make the PrivateLink endpoint available in another region to send telemetry to Datadog over PrivateLink. For more information, see the [Work With VPC Peering connections][2] page in AWS.
```

[the remark this answers is not in the document]
Comment (buraizu) [reply content/en/agent/guide/private-link.md]: Thanks for confirming, sounds good to me!

**Review by buraizu — APPROVED** [review]

**Comment by bobmshannon** [comment]
Latest changes should be pushed up and available for preview. I believe all outstanding comments have been addressed.

**Pull request merged**